tests/torch/sparsity/movement/test_algo.py

```diff
@@ -0,0 +1,794 @@
+            ffn_o=dict(weight=torch.FloatTensor([[1, 1, 0], [1, 1, 0], [1, 1, 0], [1, 1, 0]]),
+                       bias=torch.FloatTensor([1, 1, 1, 1])),
+        )
+    ),
+    'prune_1channel_no_mhsa_o_bias': dict(
+        unstructured_binary_mask=TransformerBlockItemOrderedDict(
+            mhsa_q=dict(weight=torch.FloatTensor([[0, 0, 0, 0], [0, 0, 0, 0], [1, 0, 0, 0], [1, 0, 0, 0]]),
+                        bias=torch.FloatTensor([1, 0, 0, 0])),
+            mhsa_k=dict(weight=torch.FloatTensor([[0, 0, 0, 0], [0, 0, 0, 0], [0, 1, 0, 0], [0, 0, 0, 0]]),
+                        bias=torch.FloatTensor([0, 0, 0, 0])),
+            mhsa_v=dict(weight=torch.FloatTensor([[0, 0, 0, 0], [0, 0, 0, 0], [1, 0, 0, 0], [0, 1, 0, 0]]),
+                        bias=torch.FloatTensor([0, 0, 0, 0])),
+            mhsa_o=dict(weight=torch.FloatTensor([[1, 0, 0, 1], [0, 0, 1, 0], [0, 0, 0, 0], [0, 0, 1, 0]]), bias=None),
+            ffn_i=dict(weight=torch.FloatTensor([[1, 1, 0, 1], [1, 1, 0, 1], [0, 0, 0, 0]]),
+                       bias=torch.FloatTensor([1, 0, 0])),
+            ffn_o=dict(weight=torch.FloatTensor([[0, 1, 0], [1, 1, 0], [1, 1, 0], [1, 1, 0]]),
+                       bias=torch.FloatTensor([0, 0, 0, 0])),
+        ),
+        ref_structured_binary_mask=TransformerBlockItemOrderedDict(
+            mhsa_q=dict(weight=torch.ones((4, 4)), bias=torch.ones(4)),
+            mhsa_k=dict(weight=torch.ones((4, 4)), bias=torch.ones(4)),
+            mhsa_v=dict(weight=torch.ones((4, 4)), bias=torch.ones(4)),
+            mhsa_o=dict(weight=torch.ones((4, 4)), bias=None),
+            ffn_i=dict(weight=torch.FloatTensor([[1, 1, 1, 1], [1, 1, 1, 1], [0, 0, 0, 0]]),
+                       bias=torch.FloatTensor([1, 1, 0])),
+            ffn_o=dict(weight=torch.FloatTensor([[1, 1, 0], [1, 1, 0], [1, 1, 0], [1, 1, 0]]),
+                       bias=torch.FloatTensor([1, 1, 1, 1]))
+        )
+    ),
+    'prune_none_no_ffn_bias': dict(
+        unstructured_binary_mask=TransformerBlockItemOrderedDict(
+            mhsa_q=dict(weight=torch.FloatTensor([[1, 0, 0, 0], [0, 0, 0, 0], [0, 0, 0, 0], [1, 0, 0, 0]]),
+                        bias=torch.FloatTensor([1, 0, 0, 0])),
+            mhsa_k=dict(weight=torch.FloatTensor([[0, 0, 0, 0], [0, 0, 0, 0], [0, 1, 0, 0], [0, 0, 0, 0]]),
+                        bias=torch.FloatTensor([0, 0, 0, 0])),
+            mhsa_v=dict(weight=torch.FloatTensor([[0, 0, 0, 0], [0, 0, 0, 0], [1, 0, 0, 0], [0, 1, 0, 0]]),
+                        bias=torch.FloatTensor([0, 0, 0, 0])),
+            mhsa_o=dict(weight=torch.FloatTensor([[0, 0, 0, 1], [0, 0, 1, 0], [0, 0, 0, 0], [0, 0, 1, 0]]),
+                        bias=torch.FloatTensor([1, 1, 0, 0])),
+            ffn_i=dict(weight=torch.FloatTensor([[1, 1, 0, 1], [1, 1, 0, 1], [0, 0, 0, 0]]), bias=None),
+            ffn_o=dict(weight=torch.FloatTensor([[0, 1, 0], [1, 0, 1], [1, 1, 0], [1, 1, 0]]), bias=None),
```

Comment:
The 2D tensors above could use prettier formatting, i.e. instead of:
```
            mhsa_v=dict(weight=torch.FloatTensor([[0, 0, 0, 0], [0, 0, 0, 0], [1, 0, 0, 0], [0, 1, 0, 0]]),
                        bias=torch.FloatTensor([0, 0, 0, 0])),
```
it could have looked like:
```
            mhsa_v=dict(weight=torch.FloatTensor([[0, 0, 0, 0], 
                                                  [0, 0, 0, 0], 
                                                  [1, 0, 0, 0], 
                                                  [0, 1, 0, 0]]),
                        bias=torch.FloatTensor([0, 0, 0, 0])),
```